src/sentry/grouping/enhancer/__init__.py

```diff
@@ -126,14 +126,26 @@ def keep_profiling_rules(config: str) -> str:
     return "\n".join(filtered_rules)
 
 
+class EnhancementsDict(TypedDict):
+    id: str | None
+    bases: list[str]
+    latest: bool
+    rules: NotRequired[list[EnhancementRuleDict]]
+
+
 class Enhancements:
     # NOTE: You must add a version to ``VERSIONS`` any time attributes are added
     # to this class, s.t. no enhancements lacking these attributes are loaded
     # from cache.
     # See ``GroupingConfigLoader._get_enhancements`` in src/sentry/grouping/api.py.
 
     def __init__(
-        self, rules, rust_enhancements: RustEnhancements, version=None, bases=None, id=None
+        self,
+        rules: list[EnhancementRule],
```

Comment:
Please use abstract types when within the signature (e.g. Sequence) and more specific ones when returning (e.g. list). Unless it gets on the way for whatever reason.

Reply:
This is definitely not the hill I want to die on, and I can switch it if you feel strongly. But before I go through and make a bunch of changes:

In a purely theoretical sense I guess I understand the principle, but in practice, for methods used purely internally, it always feels like overkill to me. Is there a strong argument in favor that I'm missing?

Reply:
I won't push hard on this. It is mostly as documented in the best practice document. I can understand the argument of internal usage.